core/image.cpp

```diff
@@ -725,6 +725,129 @@ static void _scale_nearest(const uint8_t *__restrict p_src, uint8_t *__restrict
 	}
 }
 
+#define LANCZOS_TYPE 3
+
+static float _lanczos(float p_x) {
+	return Math::abs(p_x) >= LANCZOS_TYPE ? 0 : Math::sincn(p_x) * Math::sincn(p_x / LANCZOS_TYPE);
+}
+
+template <int CC, class T>
+static void _scale_lanczos(const uint8_t *__restrict p_src, uint8_t *__restrict p_dst, uint32_t p_src_width, uint32_t p_src_height, uint32_t p_dst_width, uint32_t p_dst_height) {
+
+	int32_t src_width = p_src_width;
+	int32_t src_height = p_src_height;
+	int32_t dst_height = p_dst_height;
+	int32_t dst_width = p_dst_width;
+
+	float y_scale = float(p_src_height) / float(p_dst_height);
+	float x_scale = float(p_src_width) / float(p_dst_width);
+
+	float *buffer = new float[src_height * dst_width * CC]; // Store the first pass in a buffer
```

Comment:
You might want to replace the usages of `new` and `delete` with Godot's own functions.  See https://docs.godotengine.org/en/3.1/development/cpp/core_types.html#allocating-memory for info.

Reply:
Thank you, I have updated the code.